packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -603,29 +615,91 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
+
+              // Skip if URL doesn't have minimum length
+              // TODO: Could make this configurable
+              if (matchedText.length < MIN_LINK_LENGTH) {
+                continue;
+              }
+
+              const matchStart = match.index;
+              const matchEnd = matchStart + matchedText.length;
+              const urlInMatchedText = this.findURLInString(matchedText);
+              const linkMarkInRange = this.getLinkMarksInRange(
+                doc,
+                pos + matchStart,
+                pos + matchEnd,
+                true,
+              )[0];
+              const href = (urlInMatchedText && this.buildHref(urlInMatchedText.url)) || '';
+              const positionStart = pos + matchStart + (urlInMatchedText?.startIndex || 0);
+              const positionEnd = positionStart + href.length;
+
+              const unbalancedLinkUpdated = urlInMatchedText
+                ? linkMarkInRange &&
+                  !isBalanced(linkMarkInRange.text) &&
+                  isBalanced(linkMarkInRange.mark.attrs.href)
+                : false;
+
+              const outsideSelectionRange = urlInMatchedText
+                ? positionEnd < to + (from === to ? 1 : -1) ||
+                  positionStart > from - (from === to ? 1 : 0) ||
+                  (linkMarkInRange &&
+                    (linkMarkInRange.to >= to ||
+                      !within(from, linkMarkInRange.from, linkMarkInRange.to)))
+                : false;
+
+              const isExistingLinkTextNotEqualHref =
+                urlInMatchedText && linkMarkInRange
+                  ? linkMarkInRange.mark.attrs.href !== href ||
+                    linkMarkInRange.mark.attrs.href !== this.buildHref(linkMarkInRange.text)
+                  : false;
+
+              if (!urlInMatchedText) {
+                linkMarkInRange &&
+                  removeLink({ from: linkMarkInRange.from, to: linkMarkInRange.to }).tr();
+                continue;
+              }
+
+              if (isExistingLinkTextNotEqualHref) {
+                removeLink({ from: linkMarkInRange.from, to: linkMarkInRange.to }).tr();
+              }
+
+              if (
+                outsideSelectionRange &&
+                !isExistingLinkTextNotEqualHref &&
+                !unbalancedLinkUpdated
+              ) {
+                continue;
+              }
+
+              foundAutoLinks.push({
+                end: match.index + urlInMatchedText.url.length + urlInMatchedText.startIndex,
+                href,
+                start: matchStart + urlInMatchedText.startIndex,
+                text: urlInMatchedText.url,
+              });
+            }
+
+            foundAutoLinks
               // calculate link position
               .map((link) => ({
                 ...link,
                 from: pos + link.start + 1,
                 to: pos + link.end + 1,
               }))
-              .filter((link) => {
```

Comment:
The reason we moved this logic to line 650 is that we can consider changes that happen outside of the selection range that occur when balancing a URL path.